src/core.c/Rakudo/Internals.pm6

```diff
@@ -1384,15 +1384,15 @@ my class Rakudo::Internals {
             nqp::while(
               nqp::chars(my str $entry = self!next),
               nqp::stmts(
+                (next if $entry eq '.' or $entry eq '..'),
                 (my str $path = nqp::concat($!abspath,$entry)),
                 (try
                   nqp::if(
-                    $!file.ACCEPTS($entry) &&
-                      nqp::stat($path,nqp::const::STAT_ISREG),
+                      (my $s := nqp::dispatch('boot-syscall', 'file-stat', nqp::decont_s($path), 0)) && nqp::dispatch('boot-syscall', 'stat-is-reg', $s) && $!file.ACCEPTS($entry),
```

Comment:
In the final api I wonder if instead of e.g. `stat-is-reg` we should pass `nqp::const::STAT_ISREG`

Reply:
From my earlier example it looks like the file/dir stat calls return `< 1` if the file doesn't exist. Since we don't return any exceptions here to distinguish the difference between a non-existent file and a file that isn't a file or directory, we can probably get by without the existence check at all inside DIR-RECURSE

Reply:
Ah n/m, it is already doing that here

Reply:
Yeah, I was wondering about that. Should it be more like `nqp::stat` just implemented with syscalls? Or more like it is now? My WAG is that the way it is now is more spesh-able, but since we're dealing with file io, maybe not so much more as to be noticeable?